src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -662,13 +662,19 @@ STAmount::canonicalize()
         // native currency amounts should always have an offset of zero
         mIsNative = true;
 
-        if (mValue == 0)
+        // log(2^64,10) ~ 19.2
+        if (mValue == 0 || mOffset < -20)
```

Comment:
Shouldn't it be `<= -20`?

Reply:
Yes, I agree, we can exit early on equal as well. Fixed in `47030d563 [fold] Increase bounds on early exit`